src/BSON/Undefined.c

```diff
@@ -148,6 +148,7 @@ void php_phongo_undefined_init_ce(INIT_FUNC_ARGS) /* {{{ */
 	zend_class_implements(php_phongo_undefined_ce TSRMLS_CC, 1, zend_ce_serializable);
 
 	memcpy(&php_phongo_handler_undefined, phongo_get_std_object_handlers(), sizeof(zend_object_handlers));
+	php_phongo_handler_undefined.clone_obj = zend_objects_clone_obj;
```

Comment:
Although this is defined in [`std_object_handlers`](https://lxr.room11.org/xref/php-src%407.0/Zend/zend_object_handlers.c#1684), which is returned by [`zend_get_std_object_handlers`](https://lxr.room11.org/xref/php-src%407.0/Zend/zend_objects_API.c#219), I note that we _unset_ it in `phongo_get_std_object_handlers` to disable object cloning by default. That explains why this is necessary here and in the MinKey and MaxKey classes above.

Reply:
Correct. I've added a comment explaining why we reassign the standard handler.